### ISSUE TITLE
 UserAgent not being set for hybrid_remote app

### ISSUE THREAD

**Issue by mwelburn** [forcedotcom/SalesforceMobileSDK-Android#949]
I recently upgraded from 2.x to latest and noticed that the UserAgent no longer has SalesforceMobileSDK in it at all accessible via Javascript (it still seems to in the iOS build). I haven't been able to find any rationale for this outside of only being able to test in an emulator so far, but here is what navigator.userAgent outputs inside the app (as mentioned is possible here: https://developer.salesforce.com/docs/atlas.en-us.mobile_sdk.meta/mobile_sdk/hybrid_versioning.htm):

    Mozilla/5.0 (Linux; Android 5.0; Google Nexus 4 - 5.0.0 - API 21 - 768x1280 Build/LRX21M) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/37.0.0.0 Mobile Safari/537.36)

I am working on getting my hands on a physical device to see if the Genymotion emulator I am using is at fault, but for now am unsure what the cause is. For full clarity, this is a hybrid_remote app pointing to a Visualforce-based Community.

**Comment by mwelburn** [comment]
Confirmed that I still can't see a prepended SalesforceMobileSDK piece of the UserAgent on physical hardware.

**Comment by wmathurin** [comment]
Thanks for reporting this issue. We will investigate and let you know what we find out.

**Comment by mwelburn** [comment]
@wmathurin have you gotten a chance to verify whether this is still a bug or something that I might have missed?

**Comment by wmathurin** [comment]
If you use forcetk.mobilesdk.js (https://github.com/forcedotcom/SalesforceMobileSDK-Shared/blob/master/libs/forcetk.mobilesdk.js), then you should end up with the correct user agent.
Are you using it?

**Comment by mwelburn** [comment]
I am not - this is a hybrid_remote app so this is my separate web
application that needs to know if it is being invoked via a web view in an
Android app or just a standard browser. Would leveraging this javascript
file help with that (it looks like it would be loaded in the Android app
for access there rather, though without playing around I could be wrong)?

Thanks,
Michael

On Thu, Nov 12, 2015 at 5:17 PM, Wolfgang Mathurin <notifications@github.com

**Comment by mwelburn** [comment]
I am not - this is a hybrid_remote app so this is my separate web application that needs to know if it is being invoked via a web view in an Android app or just a standard browser. Would leveraging this javascript file help with that (it looks like it would be loaded in the Android app for access there rather, though without playing around I could be wrong)?

**Comment by wmathurin** [comment]
I created an hybrid remote application with forcedroid 3.3 pointing to a visualforce page that simply prints out the user agent (see below for code). And it worked as expected. We can't reproduce the bug you have reported.

VisualForce page
```html
<apex:page controller="UserAgentController" showHeader="false">
  <h1>User Agent</h1>
  {!userAgent}
</apex:page>
```
Apex controller:
```Java
public class UserAgentController {
   public String userAgent { get; set; }
   public UserAgentController() {
       userAgent = ApexPages.currentPage().getHeaders().get('USER-AGENT');
   }
}
```

**Comment by mwelburn** [comment]
It is possible at this point there is a newer release that might have fixed it (it wasn't an issue in the previous 2.3 version I had). I will investigate - thanks.

**Comment by wmathurin** [comment]
Closing for now. Re-opened, if you think there is an issue in the newest release.
